repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/library.py

```diff
@@ -248,31 +235,62 @@ def process_events(events, installed_pkgs):
     tasks = {  # Contains dicts in format {<pkg_name>: <repository>}
         'to_keep': {},
         'to_install': {},
-        'to_remove': {}}
-    for event in events:
-        if event.action in ('Deprecated', 'Present'):
-            # Add these packages to to_keep to make sure the repo they're in on RHEL 8 gets enabled
-            tasks['to_keep'].update(event.in_pkgs)
-
-        if event.action == 'Moved':
-            # Add these packages to to_keep to make sure the repo they're in on RHEL 8 gets enabled
-            # We don't care about the "in_pkgs" as it contains always just one pkg - the same as the "out" pkg
-            tasks['to_keep'].update(event.out_pkgs)
-
-        if event.action in ('Split', 'Merged', 'Renamed', 'Replaced'):
-            non_installed_out_pkgs = filter_out_installed_pkgs(event.out_pkgs, installed_pkgs)
-            tasks['to_install'].update(non_installed_out_pkgs)
-            # Add the already installed "out" pkgs to to_keep to make sure the repo they're in on RHEL 8 gets enabled
-            installed_out_pkgs = get_installed_out_pkgs(event.out_pkgs, installed_pkgs)
-            tasks['to_keep'].update(installed_out_pkgs)
-            if event.action in ('Split', 'Merged'):
-                # Uninstall those RHEL 7 pkgs that are no longer on RHEL 8
-                in_pkgs_without_out_pkgs = filter_out_out_pkgs(event.in_pkgs, event.out_pkgs)
-                tasks['to_remove'].update(in_pkgs_without_out_pkgs)
-
-        if event.action in ('Renamed', 'Replaced', 'Removed'):
-            # Uninstall those RHEL 7 pkgs that are no longer on RHEL 8
-            tasks['to_remove'].update(event.in_pkgs)
+        'to_remove': {}
+    }
+    for release in releases:
+        slated = {
+            'to_keep': {},
+            'to_install': {},
+            'to_remove': {}
+        }
+        for event in [e for e in events if e.release == release]:
+            if all(
```

Comment:
Make a function out of this five line condition. Name the function properly + a docstring, because I can't tell what is the condition supposed to do.

Reply:
Done.